filebeat/input/journald/pkg/journalread/reader.go

```diff
@@ -39,30 +39,20 @@ import (
 type Reader struct {
 	log     *logp.Logger
 	backoff backoff.Backoff
-	journal journal
+	journal *sdjournal.Journal
 }
 
 type canceler interface {
 	Done() <-chan struct{}
 	Err() error
 }
 
-type journal interface {
-	Close() error
-	Next() (uint64, error)
-	Wait(time.Duration) int
-	GetEntry() (*sdjournal.JournalEntry, error)
-	SeekHead() error
-	SeekTail() error
-	SeekCursor(string) error
-}
-
```

Comment:
The use of interfaces clouds the code. If it becomes necessary to add a mock for testing, I think it should be readded then.

Reply:
@efd6 so, are we creating additional work for a future dev who we'll be adding tests? All changes in this PR must bring value, this obviously does not.

Reply:
No, we are making the code simpler. This is an improvement. I took a look at how github.com/coreos/go-systemd/v22/sdjournal does its testing and there is no need for any mocking.

Reply:
I took a quick look at how `github.com/coreos/go-systemd/v22/sdjournal` does its testing and my understanding is that it connects to a real journald. Which would make all tests rely on `github.com/coreos/go-systemd` behaviour, which is not always describable. 

I prefer to keep the interface because it is already there and is useful for testing.

Reply:
Reluctantly done.